### DISCUSSION TITLE
SegmentedButton não foi reconhecido

### DISCUSSION THREAD

**Discussion by carolinejeunon** [Flutterando/forum#127] [Perguntas e Respostas] [answered]
Boa noite pessoal! Estou fazendo o projeto da semana fullstack e o meu SegmentedButton está com erro, e eu não consigo corrigir. Eu já fiz a importação do material... Alguém pode me ajudar por favor?

**Comment by eduardoflorence** [comment]
Bom dia @carolinejeunon, mostra o seu código do button e a mensagem de erro

**Comment by carolinejeunon** [comment]
Bom dia @eduardoflorence ! 
![image](https://user-images.githubusercontent.com/89053259/222722484-d048cf71-cf6a-4d56-9cc3-175841498c29.png)
esse é o erro, não está reconhecendo o widget. Segue meu código da página:

```
import 'package:flutter/material.dart';
import 'package:listinha/src/shared/themes/widgets/user_image_button.dart';

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('LISTINHA'),
        actions: const [
          Padding(
            padding: EdgeInsets.only(right:  8.0),
            child: UserImageButton(),
          )],
       ),
       body: Column(children: [
        SegmentedButton<int>(
          segments: [
            ButtonSegment()
          ],
          selected: const{0}
        )
       ],),
       floatingActionButton: FloatingActionButton.extended(
        onPressed: (){

        }, 
        label: const Text('Nova Lista'),
        icon: const Icon(Icons.edit),),
    );
  }
}

```

Além disso, no ThemeData, eu setei o "useMaterial3" como true, então em tese deveria funcionar. 
Ah! E a versão do Flutter que estou usando é a 3.3.10. Seria isso?

**Comment by eduardoflorence** [comment, marked as answer]
@carolinejeunon é a sua versão do Flutter. O widget SegmentedButton foi lançado na versão 3.7.0 do flutter.
É só você executar `flutter upgrade` que resolve.

**Reply by carolinejeunon** [reply]
Boa!!!! Brigada @eduardoflorence
